### ISSUE TITLE
Separate non-generated parts of client (such as REST/API client) into a package

### ISSUE THREAD

**Issue by mbohlool** [swagger-api/swagger-codegen#6191]
I am going to use python as an example, but this can be applied to most of other client libraries too.

Right now, some part of client files such as (https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/python/api_client.mustache) is not affected by input. These files are mainly responsible for providing http/auth support for API calls. 

I suggest we move them into a separate repo (swagger-base-clients) and publish them as an stand alone working client and then put a dependency on that package into generated client.

This further allow us (kubernetes) to develop our own base client with interface compatibility with swagger-base-client and switch it in kubernetes generated client.

If there is community interest in this, I can write a proposal/mock up for one language to see how it look like.

**Comment by wing328** [comment]
> Right now, some part of client files such as (https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/python/api_client.mustache) is not affected by input. These files are mainly responsible for providing http/auth support for API calls.

Yes, API client is usually the HTTP layer. For other generators like Java, the API client is swappable to support multiple HTTP libraries: Jersey 1.x & 2.x, Retrofit 1.x & 2.x, Feign, RESTEasy and more.

> I suggest we move them into a separate repo (swagger-base-clients) and publish them as an stand alone working client and then put a dependency on that package into generated client.

Yes, I think some part of the SDKs can be refactored into a separate package. For example, the [PHP ObjectSerializer](https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/php/SwaggerClient-php/lib/ObjectSerializer.php) can be a standalone PHP package.

> If there is community interest in this, I can write a proposal/mock up for one language to see how it look like.

I think we can start with the Python API client as an experiment.
